src/main/kotlin/com/google/protobuf/gradle/ProtobufConfiguratorExts.kt

```diff
@@ -4,61 +4,170 @@ import org.gradle.api.NamedDomainObjectContainer
+ *         artifact = "com.google.protobuf:protoc:3.6.1"
+ *     }
+ * }
+ * ```
+ *
+ * @receiver [ProtobufConfigurator] The protobuf plugin configuration instance
+ * for the project.
+ *
+ * @param action A lambda with receiver of type [ExecutableLocator] used
+ * for configuring the locator for protoc
+ *
+ * @return [Unit]
+ */
+fun ProtobufConfigurator.protoc(action: ExecutableLocator.() -> Unit) {
+    protoc(closureOf(action))
 }
 
-fun ProtobufConfigurator.plugins(closure: NamedDomainObjectContainerScope<ExecutableLocator>.() -> Unit) {
+fun ProtobufConfigurator.plugins(action: NamedDomainObjectContainerScope<ExecutableLocator>.() -> Unit) {
     plugins(closureOf<NamedDomainObjectContainer<ExecutableLocator>> {
-        this(closure)
+        NamedDomainObjectContainerScope.of(this).action()
     })
 }
 
-fun <T : Any> NamedDomainObjectContainerScope<T>.id(id: String, closure: (T.() -> Unit)? = null) {
-    closure?.let { create(id, closureOf(it)) }
-        ?: create(id)
-}
-
-fun <T : Any> NamedDomainObjectContainerScope<T>.remove(id: String) {
-    remove(this[id])
-}
-
-fun ProtobufConfigurator.generateProtoTasks(closure: ProtobufConfigurator.GenerateProtoTaskCollection.()->Unit) {
-    generateProtoTasks(closureOf(closure))
+fun ProtobufConfigurator.generateProtoTasks(action: ProtobufConfigurator.GenerateProtoTaskCollection.()->Unit) {
+    generateProtoTasks(closureOf(action))
 }
 
-fun GenerateProtoTask.builtins(closure: NamedDomainObjectContainerScope<GenerateProtoTask.PluginOptions>.()->Unit) {
+fun GenerateProtoTask.builtins(action: NamedDomainObjectContainerScope<GenerateProtoTask.PluginOptions>.()->Unit) {
     builtins(closureOf<NamedDomainObjectContainer<GenerateProtoTask.PluginOptions>> {
-        this(closure)
+        NamedDomainObjectContainerScope.of(this).action()
     })
 }
 
-fun GenerateProtoTask.plugins(closure: NamedDomainObjectContainerScope<GenerateProtoTask.PluginOptions>.()-> Unit) {
+fun GenerateProtoTask.plugins(action: NamedDomainObjectContainerScope<GenerateProtoTask.PluginOptions>.()-> Unit) {
     plugins(closureOf<NamedDomainObjectContainer<GenerateProtoTask.PluginOptions>> {
-        this(closure)
+        NamedDomainObjectContainerScope.of(this).action()
     })
 }
 
+/**
+ * An extension for creating and configuring the elements of an instance of [NamedDomainObjectContainer].
+ *
+ * @since 0.8.7
+ * @usage
+ * ```
+ * protobuf {
+ *     plugins {
+ *         id("grpc") {
+ *             artifact = "io.grpc:protoc-gen-grpc-java:1.15.1"
+ *         }
+ *     }
+ * }
+ * ```
+ *
+ * @receiver [NamedDomainObjectContainerScope] The scope of the [NamedDomainObjectContainer]
+ * on which to create or configure an element.
+ *
+ * @param id The string id of the element to create or configure.
+ * @param action An optional action that will be applied to the element instance.
+ *
+ * @return [Unit]
+ */
+fun <T : Any> NamedDomainObjectContainerScope<T>.id(id: String, action: (T.() -> Unit)? = null) {
```

Comment:
These two methods arent new. I just moved them down in the file and documented their usage